src/Microsoft.AspNetCore.Rewrite/Internal/PatternSegments/RewriteMapSegment.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+
+using Microsoft.AspNetCore.Rewrite.Internal.IISUrlRewrite;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Rewrite.Internal.PatternSegments
+{
+    public class RewriteMapSegment : PatternSegment
+    {
+        private readonly IISRewriteMap _rewriteMap;
+        private readonly Pattern _pattern;
+
+        public RewriteMapSegment(IISRewriteMap rewriteMap, Pattern pattern)
+        {
+            _rewriteMap = rewriteMap;
+            _pattern = pattern;
+        }
+
+        public override string Evaluate(RewriteContext context, MatchResults ruleMatch, MatchResults condMatch)
+        {
+            string key = _pattern.Evaluate(context, ruleMatch, condMatch);
+            string value;
+            if (_rewriteMap.TryGetEntry(key, out value))
+            {
+                return value;
+            }
+            var exception = new Exception($"Rewrite map entry not found: '{key}'");
```

Comment:
This style seems a bit odd. I'd refer to how we normally do exception handling with logging errors